File: src/main/java/me/kyleseven/consolereader/listeners/playerListener.java
package me.kyleseven.consolereader.listeners;

import me.kyleseven.consolereader.Utils;
import me.kyleseven.consolereader.logreader.LogAppenderManager;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

public class playerListener implements Listener {
    @EventHandler
    public void removeReadingPlayer(PlayerQuitEvent e) {
        // Remove the appender if player is offline
        if (LogAppenderManager.isReading(e.getPlayer())) {
            LogAppenderManager.stopReadingOffline(e.getPlayer());
        }
    }

    @EventHandler
    public void resumeReadingPlayer(PlayerJoinEvent e) {
        // Resume reading if the player comes back online
        if (LogAppenderManager.isReading(e.getPlayer())) {
            Utils.sendPrefixMsg(e.getPlayer(), "Console reading resumed!");
            LogAppenderManager.startReading(e.getPlayer());
        }
    }
}

File: src/main/kotlin/me/kyleseven/consolereader/listeners/PlayerListener.kt
package me.kyleseven.consolereader.listeners

import me.kyleseven.consolereader.logreader.LogAppenderManager
import me.kyleseven.consolereader.utils.sendPrefixMsg
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

class PlayerListener : Listener {
    @EventHandler
    fun removeReadingPlayer(e: PlayerQuitEvent) {
        // Preserve persistent subscriptions for the next login.
        LogAppenderManager.playerQuit(e.player)
    }

    @EventHandler
    fun resumeReadingPlayer(e: PlayerJoinEvent) {
        if (LogAppenderManager.isReadingPersistent(e.player)) {
            if (e.player.hasPermission("consolereader.read")) {
                e.player.sendPrefixMsg("Console reading resumed!")
                LogAppenderManager.startReading(e.player)
            } else {
                LogAppenderManager.stopReading(e.player)
            }
        }
    }
}
